modules/tinymce/src/themes/silver/test/ts/browser/editor/core/ChoiceControlsTest.ts

```diff
@@ -0,0 +1,301 @@
+      const hook = TinyHooks.bddSetup<Editor>({
+        ...baseSettings
+      }, [ Theme ]);
+
+      Arr.each([ menuSpec, toolbarSpec ], (spec) => {
+        it(`TINY-4843: ${spec.name} lists correct line heights`, async () => {
+          const editor = hook.editor();
+          editor.setContent('<p style="line-height: 1.4;">Hello</p>');
+          TinySelections.setCursor(editor, [ 0, 0 ], 0);
+          await spec.pOpen(editor, 'Line height');
+          await pAssertOptions(editor, spec.menuSelector, [ '1', '1.1', '1.2', '1.3', '1.4', '1.5', '2' ], Optional.some('1.4'));
+          spec.close(editor, 'Line height');
+        });
+
+        it(`TINY-4843: ${spec.name} can alter line height`, async () => {
+          const editor = hook.editor();
+          editor.setContent('<p>Hello</p>');
+          TinySelections.setCursor(editor, [ 0, 0 ], 0);
+          await spec.pOpen(editor, 'Line height');
+          await pSelectItem(editor, spec.menuSelector, '1.5');
+          TinyAssertions.assertContent(editor, '<p style="line-height: 1.5;">Hello</p>');
+        });
+
+        it(`TINY-4843: ${spec.name} only shows values within settings`, async () => {
+          const editor = hook.editor();
+          editor.setContent('<p style="line-height: 30px;">Hello</p>');
+          TinySelections.setCursor(editor, [ 0, 0 ], 0);
+          await spec.pOpen(editor, 'Line height');
+          await pAssertOptions(editor, spec.menuSelector, [ '1', '1.1', '1.2', '1.3', '1.4', '1.5', '2' ], Optional.none());
+          spec.close(editor, 'Line height');
+        });
+
+        it(`TINY-4843: ${spec.name} updates if line height changes`, async () => {
+          const editor = hook.editor();
+          editor.setContent('<p style="line-height: 1.4;">Hello</p>');
+          TinySelections.setCursor(editor, [ 0, 0 ], 0);
+          await spec.pOpen(editor, 'Line height');
+          await pAssertOptions(editor, spec.menuSelector, [ '1', '1.1', '1.2', '1.3', '1.4', '1.5', '2' ], Optional.some('1.4'));
+          editor.execCommand('LineHeight', false, '1.1');
+          await pAssertOptions(editor, spec.menuSelector, [ '1', '1.1', '1.2', '1.3', '1.4', '1.5', '2' ], Optional.some('1.1'));
+          spec.close(editor, 'Line height');
+        });
+      });
+    });
+
+    context('Float line height detection', () => {
+      const hook = TinyHooks.bddSetup<Editor>({
+        ...baseSettings,
+        lineheight_formats: '1 1.1 1.11 1.111'
+      }, [ Theme ]);
+
+      Arr.each([ menuSpec, toolbarSpec ], (spec) => {
+        it(`TINY-4843: ${spec.name} lists specified line heights`, async () => {
+          const editor = hook.editor();
+          await spec.pOpen(editor, 'Line height');
+          await pAssertOptions(editor, spec.menuSelector, [ '1', '1.1', '1.11', '1.111' ], Optional.none());
+          spec.close(editor, 'Line height');
+        });
+      });
+    });
+
+    context('Line height normalisation tests', () => {
```

Comment:
```suggestion
    context('normalisation tests', () => {
```
Otherwise this is going to read "Line height Line height normalisation tests"